### PR TITLE
linuxPackages_scx: init at 6.10

### PR DESCRIPTION
**Pull request by JohnRTitor**
## Description of changes

Description:- `sched_ext` is a Linux kernel feature which enables implementing kernel thread schedulers in BPF and dynamically loading them.
Homepage:- https://github.com/sched-ext/scx
Kernel-source:- https://github.com/sched-ext/scx-kernel-releases

This is my first time packaging a kernel, so go easy on me :)
I have mostly based this like how [bcachefs-kernel was packaged](https://github.com/NixOS/nixpkgs/blob/12c489d36b19907c6def5f73e420cd019857e2eb/pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix).

`scx` userspace tools should be inited as well, and I plan to do that in a later PR.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by JohnRTitor** [comment]
Successfully builds.

```log
❯  nom build .#linux_scx --print-out-paths
fetching git input 'git+file:///home/masum/Dev-Environment/nixpkgs'
Finished at 10:50:40 after 16s
/nix/store/q1lswsh5n2920h7qwlx5n2rxjaikkihc-linux-6.10.2-scx
```

**Comment by K900** [comment]
Can we please stop with the questionable kernel downstreams.

**Comment by JohnRTitor** [comment]
I could have got the sched-ext patches from the official repo, but they are not up-to-date.

They have the patches for `6.10.1` and it does not apply on `6.10.2` which is the current version on nixpkgs.

https://github.com/sched-ext/scx-kernel-releases

**Comment by K900** [comment]
That's not my point, my point is that we should stop shipping piles of downstream kernels, patchsets and other things, at least until we fix our kernel builder situation.

**Comment by JohnRTitor** [comment]
`sched-ext` is a promising upcoming linux-kernel feature that's probably going to be merged in next two or three linux-releases. (It was submitted for this `linux-next` cycle, but ultimately got stalled)

Adding a kernel with the patchset applied will let us test the upcoming changes and prepare for the actual release.

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 331014` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>36 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_scx.can-isotp</li>
    <li>linuxKernel.packages.linux_scx.dddvb</li>
    <li>linuxKernel.packages.linux_scx.drbd</li>
    <li>linuxKernel.packages.linux_scx.intel-speed-select</li>
    <li>linuxKernel.packages.linux_scx.ixgbevf</li>
    <li>linuxKernel.packages.linux_scx.lkrg</li>
    <li>linuxKernel.packages.linux_scx.mxu11x0</li>
    <li>linuxKernel.packages.linux_scx.ndiswrapper</li>
    <li>linuxKernel.packages.linux_scx.nvidiabl</li>
    <li>linuxKernel.packages.linux_scx.phc-intel</li>
    <li>linuxKernel.packages.linux_scx.r8125</li>
    <li>linuxKernel.packages.linux_scx.r8168</li>
    <li>linuxKernel.packages.linux_scx.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_scx.rust-out-of-tree-module</li>
    <li>linuxKernel.packages.linux_scx.tbs</li>
    <li>linuxKernel.packages.linux_scx.zfs</li>
    <li>linuxKernel.packages.linux_scx.zfs_2_1</li>
    <li>linuxKernel.packages.linux_scx.zfs_2_2</li>
    <li>linuxPackages_scx.can-isotp</li>
    <li>linuxPackages_scx.dddvb</li>
    <li>linuxPackages_scx.drbd</li>
    <li>linuxPackages_scx.intel-speed-select</li>
    <li>linuxPackages_scx.ixgbevf</li>
    <li>linuxPackages_scx.lkrg</li>
    <li>linuxPackages_scx.mxu11x0</li>
    <li>linuxPackages_scx.ndiswrapper</li>
    <li>linuxPackages_scx.nvidiabl</li>
    <li>linuxPackages_scx.phc-intel</li>
    <li>linuxPackages_scx.r8125</li>
    <li>linuxPackages_scx.r8168</li>
    <li>linuxPackages_scx.rtl8188eus-aircrack</li>
    <li>linuxPackages_scx.rust-out-of-tree-module</li>
    <li>linuxPackages_scx.tbs</li>
    <li>linuxPackages_scx.zfs</li>
    <li>linuxPackages_scx.zfs_2_1</li>
    <li>linuxPackages_scx.zfs_2_2</li>
  </ul>
</details>
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>31 packages failed to build:</summary>
  <ul>
    <li>linuxPackages_scx.ax99100 (linuxKernel.packages.linux_scx.ax99100)</li>
    <li>linuxPackages_scx.ddcci-driver (linuxKernel.packages.linux_scx.ddcci-driver)</li>
    <li>linuxPackages_scx.gasket (linuxKernel.packages.linux_scx.gasket)</li>
    <li>linuxPackages_scx.hyperv-daemons (linuxKernel.packages.linux_scx.hyperv-daemons)</li>
    <li>linuxPackages_scx.hyperv-daemons.bin (linuxKernel.packages.linux_scx.hyperv-daemons.bin)</li>
    <li>linuxPackages_scx.hyperv-daemons.lib (linuxKernel.packages.linux_scx.hyperv-daemons.lib)</li>
    <li>linuxPackages_scx.ipu6-drivers (linuxKernel.packages.linux_scx.ipu6-drivers)</li>
    <li>linuxPackages_scx.isgx (linuxKernel.packages.linux_scx.isgx)</li>
    <li>linuxPackages_scx.ithc (linuxKernel.packages.linux_scx.ithc)</li>
    <li>linuxPackages_scx.ivsc-driver (linuxKernel.packages.linux_scx.ivsc-driver)</li>
    <li>linuxPackages_scx.kvdo (linuxKernel.packages.linux_scx.kvdo)</li>
    <li>linuxPackages_scx.kvmfr (linuxKernel.packages.linux_scx.kvmfr)</li>
    <li>linuxPackages_scx.lttng-modules (linuxKernel.packages.linux_scx.lttng-modules)</li>
    <li>linuxPackages_scx.mba6x_bl (linuxKernel.packages.linux_scx.mba6x_bl)</li>
    <li>linuxPackages_scx.mbp2018-bridge-drv (linuxKernel.packages.linux_scx.mbp2018-bridge-drv)</li>
    <li>linuxPackages_scx.nvidia_x11_legacy535 (linuxKernel.packages.linux_scx.nvidia_x11_legacy535)</li>
    <li>linuxPackages_scx.nvidia_x11_legacy535.bin (linuxKernel.packages.linux_scx.nvidia_x11_legacy535.bin)</li>
    <li>linuxPackages_scx.nvidia_x11_legacy535.firmware (linuxKernel.packages.linux_scx.nvidia_x11_legacy535.firmware)</li>
    <li>linuxPackages_scx.nvidia_x11_legacy535.lib32 (linuxKernel.packages.linux_scx.nvidia_x11_legacy535.lib32)</li>
    <li>linuxPackages_scx.rtl8189es (linuxKernel.packages.linux_scx.rtl8189es)</li>
    <li>linuxPackages_scx.rtl8189fs (linuxKernel.packages.linux_scx.rtl8189fs)</li>
    <li>linuxPackages_scx.rtl8192eu (linuxKernel.packages.linux_scx.rtl8192eu)</li>
    <li>linuxPackages_scx.rtl8821au (linuxKernel.packages.linux_scx.rtl8821au)</li>
    <li>linuxPackages_scx.rtw88 (linuxKernel.packages.linux_scx.rtw88)</li>
    <li>linuxPackages_scx.shufflecake (linuxKernel.packages.linux_scx.shufflecake)</li>
    <li>linuxPackages_scx.shufflecake.bin (linuxKernel.packages.linux_scx.shufflecake.bin)</li>
    <li>linuxPackages_scx.system76-acpi (linuxKernel.packages.linux_scx.system76-acpi)</li>
    <li>linuxPackages_scx.tuxedo-keyboard (linuxKernel.packages.linux_scx.tuxedo-keyboard)</li>
    <li>linuxPackages_scx.vendor-reset (linuxKernel.packages.linux_scx.vendor-reset)</li>
    <li>linuxPackages_scx.vmm_clock (linuxKernel.packages.linux_scx.vmm_clock)</li>
    <li>linuxPackages_scx.vmware (linuxKernel.packages.linux_scx.vmware)</li>
  </ul>
</details>
<details>
  <summary>116 packages built:</summary>
  <ul>
    <li>linux_scx (linuxPackages_scx.kernel)</li>
    <li>linux_scx.dev (linuxPackages_scx.kernel.dev)</li>
    <li>linuxPackages_scx.acpi_call (linuxKernel.packages.linux_scx.acpi_call)</li>
    <li>linuxPackages_scx.akvcam (linuxKernel.packages.linux_scx.akvcam)</li>
    <li>linuxPackages_scx.apfs (linuxKernel.packages.linux_scx.apfs)</li>
    <li>linuxPackages_scx.asus-ec-sensors (linuxKernel.packages.linux_scx.asus-ec-sensors)</li>
    <li>linuxPackages_scx.asus-wmi-sensors (linuxKernel.packages.linux_scx.asus-wmi-sensors)</li>
    <li>linuxPackages_scx.batman_adv (linuxKernel.packages.linux_scx.batman_adv)</li>
    <li>linuxPackages_scx.bbswitch (linuxKernel.packages.linux_scx.bbswitch)</li>
    <li>linuxPackages_scx.bcc (linuxKernel.packages.linux_scx.bcc)</li>
    <li>linuxPackages_scx.bcc.man (linuxKernel.packages.linux_scx.bcc.man)</li>
    <li>linuxPackages_scx.bpftrace (linuxKernel.packages.linux_scx.bpftrace)</li>
    <li>linuxPackages_scx.bpftrace.man (linuxKernel.packages.linux_scx.bpftrace.man)</li>
    <li>linuxPackages_scx.broadcom_sta (linuxKernel.packages.linux_scx.broadcom_sta)</li>
    <li>linuxPackages_scx.ch9344 (linuxKernel.packages.linux_scx.ch9344)</li>
    <li>linuxPackages_scx.chipsec (linuxKernel.packages.linux_scx.chipsec)</li>
    <li>linuxPackages_scx.chipsec.dist (linuxKernel.packages.linux_scx.chipsec.dist)</li>
    <li>linuxPackages_scx.cpupower (linuxKernel.packages.linux_scx.cpupower)</li>
    <li>linuxPackages_scx.cryptodev (linuxKernel.packages.linux_scx.cryptodev)</li>
    <li>linuxPackages_scx.decklink (linuxKernel.packages.linux_scx.decklink)</li>
    <li>linuxPackages_scx.deepin-anything-module (linuxKernel.packages.linux_scx.deepin-anything-module)</li>
    <li>linuxPackages_scx.digimend (linuxKernel.packages.linux_scx.digimend)</li>
    <li>linuxPackages_scx.dpdk (linuxKernel.packages.linux_scx.dpdk)</li>
    <li>linuxPackages_scx.dpdk-kmods (linuxKernel.packages.linux_scx.dpdk-kmods)</li>
    <li>linuxPackages_scx.dpdk.doc (linuxKernel.packages.linux_scx.dpdk.doc)</li>
    <li>linuxPackages_scx.ena (linuxKernel.packages.linux_scx.ena)</li>
    <li>linuxPackages_scx.evdi (linuxKernel.packages.linux_scx.evdi)</li>
    <li>linuxPackages_scx.facetimehd (linuxKernel.packages.linux_scx.facetimehd)</li>
    <li>linuxPackages_scx.fanout (linuxKernel.packages.linux_scx.fanout)</li>
    <li>linuxPackages_scx.framework-laptop-kmod (linuxKernel.packages.linux_scx.framework-laptop-kmod)</li>
    <li>linuxPackages_scx.fwts-efi-runtime (linuxKernel.packages.linux_scx.fwts-efi-runtime)</li>
    <li>linuxPackages_scx.gcadapter-oc-kmod (linuxKernel.packages.linux_scx.gcadapter-oc-kmod)</li>
    <li>linuxPackages_scx.hid-ite8291r3 (linuxKernel.packages.linux_scx.hid-ite8291r3)</li>
    <li>linuxPackages_scx.hid-t150 (linuxKernel.packages.linux_scx.hid-t150)</li>
    <li>linuxPackages_scx.hid-tmff2 (linuxKernel.packages.linux_scx.hid-tmff2)</li>
    <li>linuxPackages_scx.hpuefi-mod (linuxKernel.packages.linux_scx.hpuefi-mod)</li>
    <li>linuxPackages_scx.it87 (linuxKernel.packages.linux_scx.it87)</li>
    <li>linuxPackages_scx.jool (linuxKernel.packages.linux_scx.jool)</li>
    <li>linuxPackages_scx.lenovo-legion-module (linuxKernel.packages.linux_scx.lenovo-legion-module)</li>
    <li>linuxPackages_scx.linux-gpib (linuxKernel.packages.linux_scx.linux-gpib)</li>
    <li>linuxPackages_scx.liquidtux (linuxKernel.packages.linux_scx.liquidtux)</li>
    <li>linuxPackages_scx.mm-tools (linuxKernel.packages.linux_scx.mm-tools)</li>
    <li>linuxPackages_scx.mstflint_access (linuxKernel.packages.linux_scx.mstflint_access)</li>
    <li>linuxPackages_scx.mwprocapture (linuxKernel.packages.linux_scx.mwprocapture)</li>
    <li>linuxPackages_scx.nct6687d (linuxKernel.packages.linux_scx.nct6687d)</li>
    <li>linuxPackages_scx.netatop (linuxKernel.packages.linux_scx.netatop)</li>
    <li>linuxPackages_scx.new-lg4ff (linuxKernel.packages.linux_scx.new-lg4ff)</li>
    <li>linuxPackages_scx.nvidia_x11 (linuxKernel.packages.linux_scx.nvidia_x11)</li>
    <li>linuxPackages_scx.nvidia_x11.bin (linuxKernel.packages.linux_scx.nvidia_x11.bin)</li>
    <li>linuxPackages_scx.nvidia_x11.firmware (linuxKernel.packages.linux_scx.nvidia_x11.firmware)</li>
    <li>linuxPackages_scx.nvidia_x11.lib32 (linuxKernel.packages.linux_scx.nvidia_x11.lib32)</li>
    <li>linuxPackages_scx.nvidia_x11_beta (linuxKernel.packages.linux_scx.nvidia_x11_beta)</li>
    <li>linuxPackages_scx.nvidia_x11_beta.bin (linuxKernel.packages.linux_scx.nvidia_x11_beta.bin)</li>
    <li>linuxPackages_scx.nvidia_x11_beta.firmware (linuxKernel.packages.linux_scx.nvidia_x11_beta.firmware)</li>
    <li>linuxPackages_scx.nvidia_x11_beta.lib32 (linuxKernel.packages.linux_scx.nvidia_x11_beta.lib32)</li>
    <li>linuxPackages_scx.nvidia_x11_beta_open (linuxKernel.packages.linux_scx.nvidia_x11_beta_open)</li>
    <li>linuxPackages_scx.nvidia_x11_production (linuxKernel.packages.linux_scx.nvidia_x11_production)</li>
    <li>linuxPackages_scx.nvidia_x11_production.bin (linuxKernel.packages.linux_scx.nvidia_x11_production.bin)</li>
    <li>linuxPackages_scx.nvidia_x11_production.firmware (linuxKernel.packages.linux_scx.nvidia_x11_production.firmware)</li>
    <li>linuxPackages_scx.nvidia_x11_production.lib32 (linuxKernel.packages.linux_scx.nvidia_x11_production.lib32)</li>
    <li>linuxPackages_scx.nvidia_x11_production_open (linuxKernel.packages.linux_scx.nvidia_x11_production_open)</li>
    <li>linuxPackages_scx.nvidia_x11_stable_open (linuxKernel.packages.linux_scx.nvidia_x11_stable_open)</li>
    <li>linuxPackages_scx.nvidia_x11_vulkan_beta (linuxKernel.packages.linux_scx.nvidia_x11_vulkan_beta)</li>
    <li>linuxPackages_scx.nvidia_x11_vulkan_beta.bin (linuxKernel.packages.linux_scx.nvidia_x11_vulkan_beta.bin)</li>
    <li>linuxPackages_scx.nvidia_x11_vulkan_beta.firmware (linuxKernel.packages.linux_scx.nvidia_x11_vulkan_beta.firmware)</li>
    <li>linuxPackages_scx.nvidia_x11_vulkan_beta.lib32 (linuxKernel.packages.linux_scx.nvidia_x11_vulkan_beta.lib32)</li>
    <li>linuxPackages_scx.nvidia_x11_vulkan_beta_open (linuxKernel.packages.linux_scx.nvidia_x11_vulkan_beta_open)</li>
    <li>linuxPackages_scx.oci-seccomp-bpf-hook (linuxKernel.packages.linux_scx.oci-seccomp-bpf-hook)</li>
    <li>linuxPackages_scx.oci-seccomp-bpf-hook.man (linuxKernel.packages.linux_scx.oci-seccomp-bpf-hook.man)</li>
    <li>linuxPackages_scx.odp-dpdk (linuxKernel.packages.linux_scx.odp-dpdk)</li>
    <li>linuxPackages_scx.openafs (linuxKernel.packages.linux_scx.openafs ,linuxPackages_scx.openafs_1_8)</li>
    <li>linuxPackages_scx.openrazer (linuxKernel.packages.linux_scx.openrazer)</li>
    <li>linuxPackages_scx.opensnitch-ebpf (linuxKernel.packages.linux_scx.opensnitch-ebpf)</li>
    <li>linuxPackages_scx.perf (linuxKernel.packages.linux_scx.perf)</li>
    <li>linuxPackages_scx.perf.debug (linuxKernel.packages.linux_scx.perf.debug)</li>
    <li>linuxPackages_scx.pktgen (linuxKernel.packages.linux_scx.pktgen)</li>
    <li>linuxPackages_scx.ply (linuxKernel.packages.linux_scx.ply)</li>
    <li>linuxPackages_scx.prl-tools (linuxKernel.packages.linux_scx.prl-tools)</li>
    <li>linuxPackages_scx.qc71_laptop (linuxKernel.packages.linux_scx.qc71_laptop)</li>
    <li>linuxPackages_scx.rr-zen_workaround (linuxKernel.packages.linux_scx.rr-zen_workaround)</li>
    <li>linuxPackages_scx.rtl8723ds (linuxKernel.packages.linux_scx.rtl8723ds)</li>
    <li>linuxPackages_scx.rtl8812au (linuxKernel.packages.linux_scx.rtl8812au)</li>
    <li>linuxPackages_scx.rtl8814au (linuxKernel.packages.linux_scx.rtl8814au)</li>
    <li>linuxPackages_scx.rtl8821ce (linuxKernel.packages.linux_scx.rtl8821ce)</li>
    <li>linuxPackages_scx.rtl8821cu (linuxKernel.packages.linux_scx.rtl8821cu)</li>
    <li>linuxPackages_scx.rtl8852au (linuxKernel.packages.linux_scx.rtl8852au)</li>
    <li>linuxPackages_scx.rtl8852bu (linuxKernel.packages.linux_scx.rtl8852bu)</li>
    <li>linuxPackages_scx.rtl88x2bu (linuxKernel.packages.linux_scx.rtl88x2bu)</li>
    <li>linuxPackages_scx.rtl88xxau-aircrack (linuxKernel.packages.linux_scx.rtl88xxau-aircrack)</li>
    <li>linuxPackages_scx.ryzen-smu (linuxKernel.packages.linux_scx.ryzen-smu)</li>
    <li>linuxPackages_scx.stdenv (linuxKernel.packages.linux_scx.stdenv)</li>
    <li>linuxPackages_scx.sysdig (linuxKernel.packages.linux_scx.sysdig)</li>
    <li>linuxPackages_scx.system76 (linuxKernel.packages.linux_scx.system76)</li>
    <li>linuxPackages_scx.system76-io (linuxKernel.packages.linux_scx.system76-io)</li>
    <li>linuxPackages_scx.system76-power (linuxKernel.packages.linux_scx.system76-power)</li>
    <li>linuxPackages_scx.system76-scheduler (linuxKernel.packages.linux_scx.system76-scheduler)</li>
    <li>linuxPackages_scx.systemtap (linuxKernel.packages.linux_scx.systemtap)</li>
    <li>linuxPackages_scx.tmon (linuxKernel.packages.linux_scx.tmon)</li>
    <li>linuxPackages_scx.tp_smapi (linuxKernel.packages.linux_scx.tp_smapi)</li>
    <li>linuxPackages_scx.trelay (linuxKernel.packages.linux_scx.trelay)</li>
    <li>linuxPackages_scx.turbostat (linuxKernel.packages.linux_scx.turbostat)</li>
    <li>linuxPackages_scx.usbip (linuxKernel.packages.linux_scx.usbip)</li>
    <li>linuxPackages_scx.v4l2loopback (linuxKernel.packages.linux_scx.v4l2loopback)</li>
    <li>linuxPackages_scx.v4l2loopback.bin (linuxKernel.packages.linux_scx.v4l2loopback.bin)</li>
    <li>linuxPackages_scx.v86d (linuxKernel.packages.linux_scx.v86d)</li>
    <li>linuxPackages_scx.veikk-linux-driver (linuxKernel.packages.linux_scx.veikk-linux-driver)</li>
    <li>linuxPackages_scx.vhba (linuxKernel.packages.linux_scx.vhba)</li>
    <li>linuxPackages_scx.virtio_vmmci (linuxKernel.packages.linux_scx.virtio_vmmci)</li>
    <li>linuxPackages_scx.virtualbox (linuxKernel.packages.linux_scx.virtualbox)</li>
    <li>linuxPackages_scx.virtualboxGuestAdditions (linuxKernel.packages.linux_scx.virtualboxGuestAdditions)</li>
    <li>linuxPackages_scx.x86_energy_perf_policy (linuxKernel.packages.linux_scx.x86_energy_perf_policy)</li>
    <li>linuxPackages_scx.xone (linuxKernel.packages.linux_scx.xone)</li>
    <li>linuxPackages_scx.xpadneo (linuxKernel.packages.linux_scx.xpadneo)</li>
    <li>linuxPackages_scx.zenpower (linuxKernel.packages.linux_scx.zenpower)</li>
    <li>linuxPackages_scx.zfs_unstable (linuxKernel.packages.linux_scx.zfs_unstable)</li>
    <li>nixpkgs-manual</li>
  </ul>
</details>

**Comment by Ma27** [comment]
The thing is, each newly added kernel causes an increased maintenance effort, especially if it's from a different vendor/upstream.

> will let us test the upcoming changes

I don't think this is sufficient to warrant the inclusion into a Linux distribution.
The bcachefs testing kernel was constantly broken and a pain to maintain for example.

> prepare for the actual release

Is there really much to do?
I'd expect at most a change in the kernel config and I don't see that much of an effort here.

In other words, second maintainer of the Linux kernel in nixpkgs :-1:ing this, hence I'd be in favor of closing this.

**Comment by AndersonTorres** [comment]
> sched-ext is a promising upcoming linux-kernel feature that's probably going to be merged in next two or three linux-releases. (It was submitted for this linux-next cycle, but ultimately got stalled)

Why not waiting, then? FOMO?
I love early-adopting things, however this is too early and, as pointed out above, too time-consuming for the current Nixpkgs staff.

> Adding a kernel with the patchset applied will let us test the upcoming changes and prepare for the actual release.

Not compelling. We can still wait the next releases, given that they will come with this feature.

**Comment by PedroHLC** [comment]
My suggestion is: instead of adding this kernel to Nixpkgs, add it to `https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/kernel/patches.nix`. Note that there is `extraConfig` field in the `{name, patch, extraConfig}`attribute sets to solidify the feature.

**Comment by JohnRTitor** [comment]
> My suggestion is: instead of adding this kernel to Nixpkgs' pkgs, add it to `https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/kernel/patches.nix`. Note that there is `extraConfig` field in the `{name, patch, extraConfig}`attribute sets to solidify the feature.

That'll also require maintenance (patch is rebased on every kernel release) and user inconvenience of rebuilding the kernel on every bump.

**Comment by JohnRTitor** [comment]
> Is there really much to do? I'd expect at most a change in the kernel config and I don't see that much of an effort here.

`scx` package (https://github.com/sched-ext/scx) needs to be added which will contain userspace schedulers such as `scx_rustland` and for that package to work a kernel with the patchset applied is needed.

A systemd service could also be designed, which will start the userspace scheduler on every boot if desired. These will be implemented under a module `nixos/scx` as `services.scx`.

**Comment by eclairevoyant** [comment]
cachyos's kernel (#330917) already includes scx support, we don't need another duplicate either, consider this the 5th 👎
